src/Nethermind/Nethermind.Experimental.Abi/AbiType.cs

```diff
@@ -0,0 +1,255 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Text;
+using Nethermind.Int256;
+
+namespace Nethermind.Experimental.Abi;
+
+public static partial class AbiType
+{
+    public static readonly IAbi<UInt256> UInt256 = new()
+    {
+        Name = "uint256",
+        Read = (ref BinarySpanReader r) =>
+        {
+            var bytes = r.ReadBytes(32);
+            return new UInt256(bytes, isBigEndian: true);
+        },
+        Write = (ref BinarySpanWriter w, UInt256 v) =>
+        {
+            Span<byte> bytes = stackalloc byte[32];
+            v.ToBigEndian(bytes);
+            w.Write(bytes);
+        },
+        Size = _ => 32,
+    };
+
+    public static readonly IAbi<UInt64> UInt64 = new()
+    {
+        Name = "uint64",
+        Read = (ref BinarySpanReader r) => (UInt64)UInt256.Read(ref r),
```

Comment:
Could we specialize these methods and read Uint64 only and have a check that the rest is 0?